### ISSUE TITLE
TypeScript resolveJsonModule  feature is broken when compiling with AOT

### ISSUE THREAD

**Issue by hervedevos** [angular/angular#25456]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
[ ] Other... Please describe:
</code></pre>

## Current behavior

When using new "resolveJsonModule" feature of TypeScript 2.9 in combination with Angular 6.1,

**ERROR in Cannot read property 'Symbol(Symbol.iterator)' of undefined** error is raised when compiling using AOT flag. Without AOT flag compilation is sucessful.

## Expected behavior

Compiling with AOT flag while using revoleJsonModul feature from TypeScript should be fine.

## Minimal reproduction of the problem with instructions

Instantiate a new project with angular-cli and add resolveJsonModule configuration to tsconfig.json

```
    "resolveJsonModule": true,
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
```

Create a json file containing an objet (e.g : src/app/test.json)

```
{
  "hello": "world"
}
```

Import this object into AppComponent

```
import json from './test.json';
```

Build Angular app with AOT:
```
ng build --aot
```

## What is the motivation / use case for changing the behavior?

Angular should be compliant with a TypeScript 2.9 feature

## Environment

<pre><code>
Angular version: 6.1.2
TypeScript version: 2.9.2

</code></pre>

**Comment by hesselbearda** [comment]
+1. Ran into the same problem as [#11931](https://github.com/angular/angular-cli/issues/11931).

It's a show stopper on our side.

**Comment by Kingwl** [comment]
i'd like to try tract this.
maybe need some help😉

**Comment by chrissena** [comment]
I'm also running into this problem. Has this worked on any version of Angular? Has anyone managed to figure out a workaround?

**Comment by hervedevos** [comment]
On my side, the only workaround I found is to deactivate AOT or to avoid use this feature

**Comment by georgiee** [comment]
Same here. I wasted quite some time on this shiny new feature. It's still totally fine to use the `typings.d.ts` approach in your angular project:

```
// file `typings.d.ts` beside your tsconfig.json
declare module "*.json" {
  const value: any;
  export default value;
}
```
This will save you the `resolveJsonModule` flag and the `allowSyntheticDefaultImports` flag. I will stick to this and don't see a reason to try the json import ever again.

**Comment by alan-agius4** [comment]
Hi,

I have tried this with Angular 7 and TypeScript 3.1 and can no longer reproduce.

**Comment by georgiee** [comment]
Just tried it and it works indeed. I immediately ran into the next issue. tslint reports any json file with several errors, most of them about the double quotes as any standard Angular setup requires single quotes in TS files. 

Reference:
https://github.com/palantir/tslint/issues/3957

**Comment by Serveny** [comment]
My solution: I used require to import the json (In my case to get the devextreme localization to work):

```
declare var require: any;
const deMessages = require('devextreme/localization/messages/de.json');
```

I know that this isn't a good solution, but I need the JSON-Import in my project only once and at the moment it's the best solution I have.

**Comment by rwoody** [comment]
@Serveny That still won't work if the values need to be used in a decorator

**Comment by pkozlowski-opensource** [comment]
Sounds like the original reported issue is already fixed and we are starting to accumulate comment for different issues here. As such this ticket is no longer actionable. If anyone faces further / similar problems please open a new issue with a minimal reproduce scenario.

**Comment by tscislo** [comment]
This also does not work on Angular 6.1.10.
Is there any way for this to work with aot on Angular 6.1.10 or should I just replace imports with require statements?
